Add SDK v5 integration steps (#794)
adjust/dev-docs: src/content/docs/sdk/ios/index.mdx

</Table>

## [4. Initialize the Adjust SDK](4-initialize-the-adjust-sdk)

To initialize the Adjust SDK, you need to create a config object. This object contains configuration options that control how the Adjust SDK behaves. Pass the following arguments for a minimal setup:

-  `appToken`: Your [app's token](https://help.adjust.com/en/article/app-settings#view-your-app-token).
-  `environment`: The environment you want to run the SDK in. Set this to `ADJEnvironmentSandbox`.

To initialize the Adjust SDK with this config object:

1. Declare your config object in the `didFinishLaunching` or `didFinishLaunchingWithOptions` method of your app delegate.
2. Set the `logLevel` property on your config object to `ADJLogLevelVerbose` (verbose). You must enable verbose logging to retrieve device information.
3. Pass the config object as an argument to the `appDidLaunch` method.

<Tabs>
<Tab title="Swift" sync="swift">

```swift
#import <AdjustSdk/Adjust.h>

> func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
>    let yourAppToken = "{YourAppToken}"
>    let environment = ADJEnvironmentSandbox as? String
>    let adjustConfig = ADJConfig(
>          appToken: yourAppToken,
>          environment: environment)
> 
>    Adjust.initSdk(adjustConfig)
>    // Your code here
>    return true
> }

Reviewer: ```suggestion
func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        // Override point for customization after application launch.
        let yourAppToken = "{YourAppToken}"
        let environment = ADJEnvironmentSandbox
        let adjustConfig = ADJConfig(appToken: yourAppToken,
                                     andEnvironment: environment)

        Adjust.initSdk(adjustConfig)
        // Your code here
        return true
    }
```

## [6. Test the Adjust SDK](6-test-the-adjust-sdk)

Now that you've configured the Adjust SDK to record information about your app, it's time to test it. Adjust offers a [testing console](https://help.adjust.com/en/article/testing-console) and a [Device API](https://help.adjust.com/en/article/device-api) to help you test your app.

Follow the [testing guide](https://help.adjust.com/en/article/test-the-adjust-sdk) to make sure Adjust receives the expected values back from your app.

## [7. Build your app for production](7-build-your-app-for-production)

Once you've finished your testing, you can build your app for production. To do this, you need to update your config object.

Update the following values:

-  `environment`: Set this to `ADJEnvironmentProduction`.
-  `logLevel`: Choose a [logging level](/en/sdk/ios/configuration#set-your-logging-level), or disable logging completely by passing an `allowSuppressLogLevel` argument in your config object.

<Tabs>
<Tab title="Swift" sync="swift">

```swift
#import <AdjustSdk/Adjust.h>

> func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
>    let yourAppToken = "{YourAppToken}"
>    let environment = ADJEnvironmentProduction as? String
>    let myAdjustConfig = ADJConfig(
>        appToken: yourAppToken,
>        environment: environment,
>        allowSuppressLogLevel: true)
>    myAdjustConfig?.logLevel = ADJLogLevelVerbose
>    //...
>    Adjust.initSdk(myAdjustConfig)
>    //...
>    return true
> }

Reviewer: ```suggestion
func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        // Override point for customization after application launch.
        let yourAppToken = "{YourAppToken}"
        let environment = ADJEnvironmentProduction
        let myAdjustConfig = ADJConfig(
            appToken: yourAppToken,
            environment: environment,
            andSuppressLogLevel: true)
        myAdjustConfig?.logLevel = ADJLogLevelVerbose
        //...
        Adjust.initSdk(myAdjustConfig)
        // Your code here
        return true
    }
```

```

</Tab>
<Tab title="Objective-C" sync="objc">

```objc
#import <AdjustSdk/Adjust.h>

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
   NSString *yourAppToken = @"{YourAppToken}";
   NSString *environment = ADJEnvironmentSandbox;
   ADJConfig *adjustConfig = [[ADJConfig alloc] initWithAppToken:appToken
                                                andEnvironment:environment];
   [adjustConfig setLogLevel:ADJLogLevelVerbose];
   //...
   [Adjust initSdk:adjustConfig];
   //...
   return YES;
}
```

</Tab>
</Tabs>

You can use `Xcode`'s build flags to dynamically update your config depending on whether you create a debug build or a production build.

<Tabs>
<Tab title="Swift" sync="swift">

```swift

> func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
>    let yourAppToken = "{YourAppToken}"
> 
>    #if DEBUG
>    let environment = ADJEnvironmentSandbox as? String
>    let adjustConfig = ADJConfig(
>          appToken: yourAppToken,
>          environment: environment)
>    adjustConfig?.logLevel = ADJLogLevelVerbose
> 
>    #else
>    let environment = ADJEnvironmentProduction as? String
>    let adjustConfig = ADJConfig(
>          appToken: yourAppToken,
>          environment: environment)
>    adjustConfig?.logLevel = ADJLogLevelSuppress
>    #endif
>    //...
>    Adjust.initSdk(adjustConfig)
>    //...
>    return true
> }

Reviewer: ```suggestion
func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        // Override point for customization after application launch.
        let yourAppToken = "{YourAppToken}"

        #if DEBUG
        let environment = ADJEnvironmentSandbox
        let adjustConfig = ADJConfig(
            appToken: yourAppToken,
            andEnvironment: environment)
        adjustConfig?.logLevel = ADJLogLevelVerbose

        #else
        let environment = ADJEnvironmentProduction
        let adjustConfig = ADJConfig(
            appToken: yourAppToken,
            andEnvironment: environment)
        adjustConfig?.logLevel = ADJLogLevelSuppress
        #endif
        //...
        Adjust.initSdk(adjustConfig)
        //...
        return true
}
```

```

</Tab>
<Tab title="Objective-C" sync="objc">

```objc
#import "Adjust.h"

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
   NSString *yourAppToken = @"{YourAppToken}";

   #if DEBUG
   NSString *environment = ADJEnvironmentSandbox;
   ADJConfig *adjustConfig = [[ADJConfig alloc] initWithAppToken:appToken
                                                andEnvironment:environment];
   [adjustConfig setLogLevel:ADJLogLevelVerbose];

   #else
   NSString *environment = ADJEnvironmentProduction;
   ADJConfig *adjustConfig = [[ADJConfig alloc] initWithAppToken:appToken
                                                andEnvironment:environment];
   [adjustConfig setLogLevel:ADJLogLevelSuppress];
   #endif
   //...

>    [Adjust initSdk:myAdjustConfig];

Reviewer: ```suggestion
   [Adjust initSdk: adjustConfig];
```

| `WebKit.framework`                  | Enables the use of web views in your application                                                                                 | Only required if your app uses web views                                                                        |

</Table>

## [4. Initialize the Adjust SDK](4-initialize-the-adjust-sdk)

To initialize the Adjust SDK, you need to create a config object. This object contains configuration options that control how the Adjust SDK behaves. Pass the following arguments for a minimal setup:

-  `appToken`: Your [app's token](https://help.adjust.com/en/article/app-settings#view-your-app-token).
-  `environment`: The environment you want to run the SDK in. Set this to `ADJEnvironmentSandbox`.

To initialize the Adjust SDK with this config object:

1. Declare your config object in the `didFinishLaunching` or `didFinishLaunchingWithOptions` method of your app delegate.
2. Set the `logLevel` property on your config object to `ADJLogLevelVerbose` (verbose). You must enable verbose logging to retrieve device information.

> 3. Pass the config object as an argument to the `appDidLaunch` method.

Reviewer: ```suggestion
3. Pass the config object as an argument to the `initSdk` method.
```
